src/python/pants/backend/python/goals/lockfile.py

```diff
@@ -275,18 +276,28 @@ async def python_lockfile_synthetic_targets(
         (os.path.dirname(lockfile), os.path.basename(lockfile), name)
         for name, lockfile in python_setup.resolves.items()
     ]
+
+    def synthesize_BUILD_file(
+        spec_path: str, lockfiles: Iterator[tuple[str, str, str]]
+    ) -> tuple[str, tuple[TargetAdaptor, ...]]:
+        source = os.path.join(spec_path, "BUILD.python-lockfiles")
+        return (
+            source,
+            tuple(
+                TargetAdaptor(
+                    "_lockfiles",
+                    name=synthetic_lockfile_target_name(name),
+                    __source__=(source, 0),
+                    sources=[lockfile],
+                )
+                for _, lockfile, name in lockfiles
+            ),
+        )
```

Comment:
Rather than calling this field `build_file_source`, why not something like "description_of_origin", to follow the pattern that we use elsewhere for error messages? That would make it more obvious in this case that what you want in the field is something like:
> "the `[python].resolves` option"

... in this case.

Reply:
eh.. wait what?

I don't want `"the [python].resolves option"` here.. so I'm not sure I follow what you're suggesting.

Reply:
I do see `build_file_source` as a form of "description of origin" for where the target comes from (is declared), but that will always be a path to a BUILD file and a line number (which may be set to 0 for syntehtic targets)

Reply:
But the build file name for synthetic targets is already made up... they definitely do not come from any particular BUILD file, so there is no point having a particular filename there.

Reply:
With the synthetic target work, do we have the concept of a synthetic build file persisted anywhere as well? Or it the synthetic target just tucked away in memory somewhere?

For example, in SvelteKit, we have something like:

`.svelte-kit/types/.../$types.d.ts` or `svelte-kit/generated/root.svelte`

Those are generated on first run, or when something changes. Great for IDE integration and for manual inspection.

Not sure if we want to do something similar - which then would make the "build_file_source" make sense again :)

Reply:
> But the build file name for synthetic targets is already made up... they definitely do not come from any particular BUILD file, so there is no point having a particular filename there.

True, except we use the made up filename in error messages already and there is still an assumption in general that targets are declared in actual files so I don't follow the reasoning that synthetic targets should invalidate the filename property for targets, at least for now.

Just to be clear, this is not just for synthetic targets.. it's a generic change to all targets that happens to impact synthetic targets too.


> Those are generated on first run, or when something changes. Great for IDE integration and for manual inspection.
> 
> Not sure if we want to do something similar - which then would make the "build_file_source" make sense again :)

That's an interesting concept, currently it is all in memory. What if `peek` had an alternative output format like `--output-format=BUILD` that you could use for any targets you care about (including synthetic ones)..

Reply:
Big but related tangent: `buck2` has an interesting variant of synthetic targets: ["anonymous targets"](https://buck2.build/docs/rule_authors/anon_targets/), which don't actually have a name, and it instead use the fingerprint of their fields as their identity in the graph. I agree with their premise that even attempting to uniquely name these targets isn't a good idea (just their arguments and type are enough).

Reply:
> Just to be clear, this is not just for synthetic targets.. it's a generic change to all targets that happens to impact synthetic targets too.

I understand that. I think that to actually make it generic though, you would want to acknowledge that not all targets have a BUILD file or a line number, and instead make this field generic.

Reply:
renamed attr to `description_of_origin`.